libs/variant/include/variant/variant.hpp

```diff
@@ -576,4 +584,150 @@ void Variant::IterateObject(Function const &function) const
 }
 
 }  // namespace variant
+
+namespace serializers {
+
+template <typename D>
+struct ForwardSerializer<fetch::variant::Variant, D>
+{
+public:
+  using Type       = fetch::variant::Variant;
+  using DriverType = D;
+
+  template <typename Serializer>
+  static void Serialize(Serializer &serializer, Type const &var)
+  {
+    auto typecode = static_cast<int>(var.type_);
+    serializer << typecode;
+    switch (var.type_)
+    {
+    case Type::Type::UNDEFINED:
+    {
+      break;
+    }
+
+    case Type::Type::NULL_VALUE:
+    {
+      break;
+    }
+    case Type::Type::INTEGER:
+    {
+      serializer << var.primitive_.integer;
+      break;
+    }
+    case Type::Type::FLOATING_POINT:
+    {
+      serializer << var.primitive_.float_point;
+      break;
+    }
+    case Type::Type::FIXED_POINT:
+    {
+      serializer << var.primitive_.integer;
+      break;
+    }
+    case Type::Type::BOOLEAN:
+    {
+      serializer << var.primitive_.boolean;
+      break;
+    }
+    case Type::Type::STRING:
+    {
+      serializer << var.string_;
+      break;
+    }
+    default:
+    {
+      throw std::runtime_error{"Variant has unknown type."};
+    }
```

Comment:
should be relocated after the switch(...){...} block and change break; to return;

Reply:
done